File: server/InternshipPlatformAPI/InternshipPlatformAPI/Models/Application.cs
using System.ComponentModel.DataAnnotations;

namespace InternshipPlatformAPI.Models
{
    public class Application
    {
        public Guid Id { get; set; } = Guid.NewGuid();
        [Required]
        public string FirstName { get; set; } = string.Empty;
        [Required]
        public string LastName { get; set; } = string.Empty;
        [EmailAddress]
        public string Email { get; set; } = string.Empty;
        [Required]
        public string EducationLevel { get; set; } = string.Empty;
        [Required]
        [MaxLength(300)]
        public string CoverLetter { get; set; } = string.Empty;
        [Required]
        public string CV { get; set; } = string.Empty;
        [Required]
        public string Status { get; set; } = "applied";
     
        public ICollection<ApplicationComment>? Comments { get; set; }
       
        public ICollection<Selection>? Selections { get; set; }
    }
}

File: client/src/app/models/Application.ts
// export interface Application {
//   Id: number;
//   FirstName: string;
//   LastName: string;
//   Email: string;
//   EducationLevel: string;
//   CoverLetter: string;
//   CV: string;
//   Status: string;
// }
export class Application {
  constructor(
    public Id: number,
    public firstName: string,
    public lastName: string,
    public Email: string,
    public EducationLevel: string,
    public CoverLetter: string,
    public CV: string,
    public Status: string
  ) {}
}
